src/slave/slave.cpp

```diff
@@ -3654,6 +3655,32 @@ void Slave::launchExecutor(
       authenticationToken.get(),
       framework->info.checkpoint());
 
+
+  if(environment.find("CRITEO_CNI") != environment.end() && environment["CRITEO_CNI"] == "enabled"){
+
+    // Slave has CNI enabled but doesn't define any networks, public and private subnets must be injected.
+    ContainerInfo *containerInfo = containerConfig.mutable_container_info();
+
+    NetworkInfo *privateNet = containerInfo->add_network_infos();
+    NetworkInfo *publicNet = containerInfo->add_network_infos();
+    
+    publicNet->set_name("public");
+    privateNet->set_name("private");
+
+
+    Ports ports = *containerConfig.mutable_executor_info()->mutable_discovery()->mutable_ports();
+
+    // We need to take the defined ports and map them in the public subnet
+    for(const auto port : *ports.mutable_ports()){
+
+        auto p = publicNet->add_port_mappings();
+        p->set_host_port(port.number());
+        p->set_container_port(port.number());
+        p->set_protocol(port.protocol());
+
```

Comment:
I think you need to manage other flags too? We saw that users can also set "labels"; does it need to be reported here too?

Reply:
no the labels aren't included here. they're also sent correctly anyway